toxic-review-warning/src/script.js

```diff
@@ -0,0 +1,131 @@
+const url = new URL('./worker.js', import.meta.url);
+import { MESSAGE_CODE, MODEL_STATUS } from './consts.js';
+import { TYPING_DELAY } from './config.js';
+
+// Initialization
+
+let modelStatus = MODEL_STATUS.NOT_STARTED;
+let isToxicityVisible = false;
+updateToxicityVisibility(isToxicityVisible);
+displayModelStatus(modelStatus);
+updateToxicityAssessmentDisplay({ isToxic: false, toxicityTypeList: [] });
+const worker = new Worker(url);
+let typingTimeout = 0;
+
+// DOM manipulation, display and events
+
+function updateToxicityVisibility(isVisible) {
+  isToxicityVisible = isVisible;
+  document.getElementById('aiWrapper').className = isToxicityVisible
+    ? 'visible'
+    : 'hidden';
+}
+
+function handleUserInputChange(event) {
+  if (isToxicityVisible === true) {
+    updateToxicityVisibility(false);
+  }
+  if (typingTimeout) {
+    clearTimeout(typingTimeout);
+  }
+  typingTimeout = setTimeout(() => {
+    console.info('Input: User has stopped typing');
+    if (modelStatus === MODEL_STATUS.READY) {
+      console.info('Trigger inference');
+      runLLMInference();
+    }
+    setTimeout(() => {}, 1000);
+  }, TYPING_DELAY); // 800ms delay
```

Comment:
```suggestion
  }, TYPING_DELAY);
```

If you want to comment on a delay that works well in practice, do it where the delay is set.